mozilla/pontoon
# Add light theme CSS file

Fix #2935. In addition to adding the light theme, this patch also includes consistency changes on the dark theme.

**Note**: after deployment, a small change is required to the homepage contents stored in the database, because a Mozilla logo in the footer has been moved from HTML to CSS. I've already done it on stage to where this patch is currently deployed.

**How to test themes in Pontoon?**
1. Go to https://mozilla-pontoon-staging.herokuapp.com/settings/ and select Light theme.
2. Explore Pontoon and comment here if you notice anything unusual. See a list of pages to check below.
3. Go to [Settings](https://mozilla-pontoon-staging.herokuapp.com/settings/) again, select Dark theme and repeat the test. We've also made minor changes to the Dark theme.

**What pages to test?**

1. Dashboards:
- https://mozilla-pontoon-staging.herokuapp.com/projects/
- https://mozilla-pontoon-staging.herokuapp.com/projects/firefox/
- https://mozilla-pontoon-staging.herokuapp.com/teams/
- https://mozilla-pontoon-staging.herokuapp.com/fr/
- https://mozilla-pontoon-staging.herokuapp.com/fr/firefox/

2. User pages:
- https://mozilla-pontoon-staging.herokuapp.com/contributors/
- https://mozilla-pontoon-staging.herokuapp.com/profile/
- https://mozilla-pontoon-staging.herokuapp.com/settings/
- https://mozilla-pontoon-staging.herokuapp.com/notifications/

3. Admin:
- https://mozilla-pontoon-staging.herokuapp.com/admin/
- https://mozilla-pontoon-staging.herokuapp.com/admin/projects/firefox/
- https://mozilla-pontoon-staging.herokuapp.com/admin/projects/tutorial/strings/

4. Other assets:
- https://mozilla-pontoon-staging.herokuapp.com/machinery/
- https://mozilla-pontoon-staging.herokuapp.com/terms/
- https://mozilla-pontoon-staging.herokuapp.com/insights/
- https://mozilla-pontoon-staging.herokuapp.com/sync/
- https://mozilla-pontoon-staging.herokuapp.com/ (in [Settings](https://mozilla-pontoon-staging.herokuapp.com/settings/) select "Default Homepage" as your Homepage)
- Add-On Promotion banner

5. Translate View:
* https://mozilla-pontoon-staging.herokuapp.com/fr/firefox/all-resources/

**Known blocking issues:**

- [ ] Theme toggle in the Profile Menu of the Translate View is not yet available
- [ ] [FOUC](https://en.wikipedia.org/wiki/Flash_of_unstyled_content) when Match system is used and the system theme is Light
- [x] In Light theme, Translated status color doesn't have sufficient contrast when used `with font-weight: 300` ([more](https://github.com/mozilla/pontoon/pull/2997#issuecomment-1775273074))
- [x] In Light theme, Pretranslated status color has inappropriate contrast in dashboard progress bar hover
- [x] In Light theme, Missing status color is too dark
- [x] In Light theme, Translation Time Filter colors are too light, because they are hardcoded
- [x] In Light theme, Machinery diff and String list/Concordance search highlights are too dark
- [x] In Dark theme, Priority star in heading needs more contrast
- [x] In Light theme, (manually sent) Notification message color needs more contrast
- [x] Delete button style in the Database project String admin is broken

**Known non-blocking issues:**

- [ ] Interactive Tour colors are hardcoded (light)
- [ ] https://ui-avatars.com/ (shown if no Gravatar image found) colors are hardcoded (dark)
- [ ] Separate main border in two (too dark when between dark and light background on dark theme)
- [ ] Light-theme: Insights chart backgrounds don't work too well with the light theme

**Issues not related to this patch (i.e. issues to be filed):**
- [ ] Buttons in translate view have borders and inputs don't (it's the opposite elsewhere)
- [ ] Welcome to Locale X panel is invisible
- [ ] De-duplicate Notification, Profile menu and Sign in buttons in the header (use React on Dashboards)
